src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1148,6 +1156,30 @@ private void retrieveChanges(Run build, GitClient git, TaskListener listener) th
         }
     }
 
+    private boolean determineRedundantFetch(CloneOption option, RemoteConfig rc) {
+        List<RefSpec> initialFetchRefSpecs = rc.getFetchRefSpecs();
+        boolean isDefaultRefspec = true; // default refspec is any refspec with "refs/heads/" mapping
+        boolean removeSecondFetch = true;
+        if (initialFetchRefSpecs != null) {
+            for (RefSpec ref:initialFetchRefSpecs) {
```

Comment:
For clarity and consistency with other foreach loops
```suggestion
            for (RefSpec ref : initialFetchRefSpecs) {
```

Reply:
Thanks @fcojfernandez !